api/backend.proto

```diff
@@ -17,62 +17,76 @@ package api;
 option go_package = "internal/future/pb";
 
 import 'api/messages.proto';
+import 'google/api/annotations.proto';
 
-// Configuration for the Match Function to be triggered by Open Match to generate proposals.
+// Configuration for the Match Function to be triggered by Open Match to
+// generate proposals.
 message MatchFunctionConfig {
   message GrpcFunctionProperties {
-      string host = 1;
-      int32 port = 2;
+    string host = 1;
+    int32 port = 2;
   }
 
   message RestFunctionProperties {
-      string host = 1;
-      int32 port = 2;
+    string host = 1;
+    int32 port = 2;
   }
 
   // A developer-chosen human-readable name for this Match Function.
   string name = 1;
 
   // Properties for the type of this function.
   oneof function {
-    GrpcFunctionProperties grpc_function = 101;
-    RestFunctionProperties rest_function = 102;
+    GrpcFunctionProperties grpc_function = 10001;
+    RestFunctionProperties rest_function = 10002;
   }
 }
 
-message GetMatchesRequest{
+message GetMatchesRequest {
   // MatchFunction to be executed for the given list of MatchProfiles
   MatchFunctionConfig function = 1;
 
   // MatchProfiles for which this MatchFunction should be executed.
-  repeated messages.MatchProfile profile = 2;
+  repeated MatchProfile profile = 2;
 }
 
-message GetMatchesResponse{
+message GetMatchesResponse {
   // Result Match for the requested MatchProfile.
-  messages.Match match = 1;
+  Match match = 1;
 }
 
 message AssignTicketsRequest {
   // List of Ticket ids for which the Assinment is to be made.
   repeated string ticket_id = 1;
 
   // Assignment to be associated with the Ticket ids.
-  messages.Assignment assignment = 2;
+  Assignment assignment = 2;
 }
 
-message AssignTicketsResponse {
-}
+message AssignTicketsResponse {}
 
-// The service implementing the Backent API that is called to generate matches and make assignments for Tickets.
+// The service implementing the Backent API that is called to generate matches
+// and make assignments for Tickets.
 service Backend {
-  // GetMatches triggers execution of the specfied MatchFunction for each of the specified MatchProfiles.
-  // Each MatchFunction execution returns a set of proposals which are then evaluated to generate results.
-  // GetMatches method streams these results back to the caller.
+  // GetMatches triggers execution of the specfied MatchFunction for each of the
+  // specified MatchProfiles. Each MatchFunction execution returns a set of
+  // proposals which are then evaluated to generate results. GetMatches method
+  // streams these results back to the caller.
+  // TODO: Should this be renamed to createProposal? It's not a "Get" if it's
```

Comment:
I can see why 'get' might not be a good idea (since matches get generated - not simply fetched)

This method does not cause any state change in Open Match (now that proposals, matches are not stored in state storage) - no Open Match stored resource is created here. Is 'create' still a write verb then? Or would a custom verb (say generatematches / makematches / ... be more appropriate?)

Reply:
I do not want to redesign the API in this change. Let's leave the TODO as is until reviewed. There were some things I had to workaround. In this case you cannot have a GET verb with a body so I have to use POST which cascades into requiring a verb name since a POST against a collection is an insert of that resource.